### PR TITLE
MDCT-1962: Modify ChoiceList choices to store name and value

### PR DESCRIPTION
**Pull request by ntsummers1**
## Description
Fixes [MDCT-1962](https://qmacbis.atlassian.net/browse/MDCT-1962) Choices have even more meaning

https://user-images.githubusercontent.com/19439679/191330006-5158ad19-6f0f-4fe0-9da4-a99165eeda54.mov

### How to test
./dev local
Create a program
Go to a page with radio buttons or checkboxs
Click and enter info to your hearts content
Save the form
Come back and make sure its hydrated!

## Code author checklist
- [ ] I have performed a self-review of my code
- [ ] I have added [thorough](https://bit.ly/3zPrxuZ) tests
- [ ] I have added analytics, if necessary
- [ ] I have updated the documentation, if necessary

## Reviewer checklist (two different people)
- [ ] I have done the deep review and verified the items in the above checklist are g2g
- [ ] I have done the lgtm review

### CODE REVIEW

File: services/ui-src/src/components/fields/ChoiceListField.tsx [hunk 1]
```diff
@@ -55,9 +56,13 @@ export const ChoiceListField = ({
...
       form.setValue(name, displayValue, { shouldValidate: true });
     }
+
     // update DOM choices checked status
     choices.forEach((choice: FieldChoice) => {
-      choice.checked = displayValue?.includes(choice.value) || false;
+      const foundValue = displayValue?.find(
+        (value) => value.value === choice.value
+      );
+      choice.checked = !!foundValue;
```
Comment (braxex) [inline services/ui-src/src/components/fields/ChoiceListField.tsx:65]: Nit: I found this method a little hard to parse due to all the `value`. Maybe:
```suggestion
      const checkedState = displayValue?.find(
        (option) => option.value === choice.value
      );
      choice.checked = !!checkedState;
```

File: services/ui-src/src/components/fields/ChoiceListField.tsx [hunk 2]
```diff
@@ -81,7 +86,7 @@ export const ChoiceListField = ({
 
   // update field values
   const onChangeHandler = (event: InputChangeEvent) => {
-    const clickedOption = event.target.value;
+    const clickedOption = { key: event.target.name, value: event.target.value };
```
[the remark this answers is not in the document]
Comment (ntsummers1) [reply services/ui-src/src/components/fields/ChoiceListField.tsx:89]: What do you mean by optionName? We have this element here:
```
<input class="ds-c-choice" id="checkbox_bedr-1_121" name="bedr-1" type="checkbox" value="Other state agency staff">
```
Do you mean using the ID instead?

**Review by braxex — APPROVED** [review]
🚢 it

**Review by gmrabian — CHANGES_REQUESTED** [review]
I'm getting an error when I try to create a program with the checkbox selected. It works when it's not.

https://user-images.githubusercontent.com/57802560/191603664-c646c025-5c2d-46e2-93db-29f66b1a0622.mov

**Comment by ntsummers1** [comment]
@gmrabian Thanks for pointing that out! Fixed it, and while I was at it I went ahead and hooked that little checkbox up so now it saves whether or not the user checked it or not and will reflect that upon trying to edit a program.

**Review by braxex — APPROVED** [review]
@ntsummers1 @gmrabian These recent changes make sense to me, but I wonder if it would be easier and cleaner to just pass a boolean value for `combinedData` instead. It's really only ever going to be `true` or `false`, and because it's just one optional checkbox, we can use the `singleCheckbox` validation for it.

Also @ntsummers1 because this solves the [AddEditProgramModal checkbox hydration ticket](https://qmacbis.atlassian.net/browse/MDCT-1955), I've assigned that to you. 😄

**Review by gmrabian — CHANGES_REQUESTED** [review]
Good change, though now we have the issue where it's always on once you check it once 😅 

https://user-images.githubusercontent.com/57802560/191758947-5b2a6fdc-994c-4833-be08-3ec9c46b92ab.mov

**Comment by ntsummers1** [comment]
> @ntsummers1 @gmrabian These recent changes make sense to me, but I wonder if it would be easier and cleaner to just pass a boolean value for `combinedData` instead. It's really only ever going to be `true` or `false`, and because it's just one optional checkbox, we can use the `singleCheckbox` validation for it.
> 
> 
> 
> Also @ntsummers1 because this solves the [AddEditProgramModal checkbox hydration ticket](https://qmacbis.atlassian.net/browse/MDCT-1955), I've assigned that to you. 😄 

I'll see what I can do!

**Review by gmrabian — APPROVED** [review]
looks good! Thanks for fixing that

**Review by gmrabian — APPROVED** [review]
good!

**Review by gmrabian — APPROVED** [review]
